uninstall.sh

```diff
@@ -18,6 +18,7 @@ sleep 2
 
 ## now check what shell is running
 shell_type=$(basename "$SHELL")
+# shellcheck disable=SC2034
```

Comment:
What's the unused variable here?

Reply:
I think this is more that the specific variable is unknown, rather than undefined. The presence of `?` would seem to indicate that the shell being used is not one of the known types and could be something else entirely.

Looking at the [documentation][] - at least to my eyes - this looked like one of the exceptional conditions which produces a false warning, so as long as the script still works as intended, it can be ignored.

[documentation]: https://github.com/koalaman/shellcheck/wiki/SC2034#exceptions